readthedocs/templates/builds/build_detail.html

```diff
@@ -69,6 +69,23 @@
           </a>
         </li>
       </div>
+
+
+      {# Show rebuild button only if the version is external and it's the latest build for this version #}
+      {# see https://github.com/readthedocs/readthedocs.org/pull/6995#issuecomment-852918969 #}
+      {% if request.user|is_admin:project and build.version.type == "external" and is_latest_build %}
+      <div data-bind="visible: finished()">
+        <form method="post" name="rebuild_commit" action="{% url "builds_project_list" project.slug %}">
```

Comment:
Feels weird having this on the same list view, I think we could have another view `/builds/{pk}/trigger/` or similar.

Reply:
I think this is a common pattern: POST to a list view to create a new object. We are using this in the same way when triggering a build for a particular version on Build List page at https://github.com/readthedocs/readthedocs.org/blob/2f9244ee3fb2e826255954a5aa4ff694ff7941e4/readthedocs/templates/builds/build_list.html#L33

Seems fine to me to re-use the view and just pass `build_pk` to trigger a build for a particular version at a specific point in time.